dns/client.go

```diff
@@ -102,13 +148,141 @@ func (r *Client) Resolve(
 	return ips, nil
 }
 
-// Lookup resolves a domain name to a slice of IP addresses using the system's
-// default resolver.
+// Lookup resolves a domain name to a slice of IP addresses.
+//
+// It prefers the k6 VU dialer's resolver if available, so that k6 networking
+// options (e.g. blockHostnames) are honored. It falls back to the system's
+// default resolver otherwise.
 func (r *Client) Lookup(ctx context.Context, hostname string) ([]string, error) {
-	ips, err := net.DefaultResolver.LookupHost(ctx, hostname)
+	vustate := r.vu.State()
+	if vustate == nil {
+		return nil, common.NewInitContextError("using lookup in the init context is not supported")
```

Comment:
Out of curiosity, is `Resolve` permitted while `Lookup` isn't, for any particular reason?

Reply:
Should we have a centralized function and delegate this check there?